app/src/main/java/com/humberto/tasky/agenda/presentation/components/AgendaListItem.kt

```diff
@@ -0,0 +1,165 @@
+                        style = MaterialTheme.typography.headlineMedium.copy(
+                            fontWeight = FontWeight.W600,
+                            textDecoration =
+                                if (agendaItemUi.isItemChecked == true) TextDecoration.LineThrough
+                                else TextDecoration.None
+                        ),
+                        maxLines = 1,
+                        overflow = TextOverflow.Ellipsis
+                    )
+                    Spacer(modifier = Modifier.height(8.dp))
+                    Text(
+                        text = agendaItemUi.description,
+                        color = cardStyle.descriptionColor,
+                        style = MaterialTheme.typography.bodyLarge,
+                        minLines = 2,
+                        maxLines = 2,
+                        overflow = TextOverflow.Ellipsis,
+                    )
+                }
+            }
+            MoreButtonWithDropDownMenu(
+                menuItems = listOf(
+                    DropDownItem(
+                        title = stringResource(id = R.string.open),
+                        onClick = onOpenItem,
+                    ),
+                    DropDownItem(
+                        title = stringResource(id = R.string.edit),
+                        onClick = onEditItem,
+                    ),
+                    DropDownItem(
+                        title = stringResource(id = R.string.delete),
+                        onClick = onDeleteItem,
+                    ),
+                ),
+                iconColor = cardStyle.moreButtonColor,
+            )
+        }
+        Text(
+            modifier = Modifier.fillMaxWidth(),
+            color = cardStyle.descriptionColor,
+            textAlign = TextAlign.End,
+            text = agendaItemUi.dateTime,
+            style = MaterialTheme.typography.bodyLarge
+        )
+    }
+}
+
+@Composable
+private fun getCardStyle(eventType: EventType): CardStyle {
+    return when (eventType) {
+        is EventType.Task -> CardStyle(
+            backgroundColor = MaterialTheme.colorScheme.secondary,
+            titleColor = MaterialTheme.colorScheme.onPrimary,
+            descriptionColor = MaterialTheme.colorScheme.onSecondary,
+            moreButtonColor = MaterialTheme.colorScheme.onSecondary
+        )
+        is EventType.Event -> CardStyle(
+            backgroundColor = TaskyLightGreen,
+            titleColor = MaterialTheme.colorScheme.onSurface,
+            descriptionColor = MaterialTheme.colorScheme.onSurfaceVariant,
+            moreButtonColor = TaskyBrown
+        )
+        is EventType.Reminder -> CardStyle(
+            backgroundColor = MaterialTheme.colorScheme.surfaceVariant,
+            titleColor = MaterialTheme.colorScheme.onSurface,
+            descriptionColor = MaterialTheme.colorScheme.onSurfaceVariant,
+            moreButtonColor = TaskyBrown
+        )
+    }
+}
```

Comment:
Personally, I would simplify this, move this logic to the composable, and assign the properties to variables. It makes sense to contain it there because that is where it is needed. I also avoid maintaining a separate class and Composable, and I find it easier to read.

```
val backgroundColor = when (item.type) {
  AgendaItemType.EVENT -> TaskyLightGreen
  AgendaItemType.TASK -> TaskyGreen
  AgendaItemType.REMINDER -> TaskyLight2
}

val foregroundColor = when (item.type) {
  AgendaItemType.TASK -> TaskyWhite
  else -> TaskyBlack
}

val textColor = when (item.type) {
  AgendaItemType.TASK -> TaskyWhite
  else -> TaskyDarkGray
}
```